### PR TITLE
Add update_usage.py script

### PR DESCRIPTION
**Pull request by Avgor46**
Added a script for convenient `docs/usage.md` updating.

### CODE REVIEW

File: update_usage.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+command = subprocess.run(["cargo", "build"])
+if command.returncode != 0:
+    print('Cargo build exited with code ' + command.returncode)
+
+build_dir = os.path.dirname(os.path.realpath(__file__))
+target_dir = os.path.join(build_dir, "target", "debug")
+
+for p in os.listdir(target_dir):
+    if p.startswith("casr-") and os.path.isfile(target := os.path.join(target_dir, p)) \
```
Comment (hkctkuy) [inline update_usage.py:18]: Maybe replace the if condition with its opposite and add `continue` as it done below for better code readability?

[the remark this answers is not in the document]
Comment (Avgor46) [reply update_usage.py]: done

Comment (Avgor46) [reply update_usage.py]: I think, nesting level is acceptable

File: update_usage.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    print('Cargo build exited with code ' + command.returncode)
+
+build_dir = os.path.dirname(os.path.realpath(__file__))
+target_dir = os.path.join(build_dir, "target", "debug")
+
+for p in os.listdir(target_dir):
+    if p.startswith("casr-") \
+            and os.path.isfile(target := os.path.join(target_dir, p)) \
+            and os.access(target, os.X_OK):
+
```
Comment (hkctkuy) [inline update_usage.py:21]: Remove empty line

Comment (Avgor46) [reply update_usage.py:21]: It was added for readability.

File: update_usage.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
+#! /usr/bin/env python3
```
Comment (SweetVishnya) [inline update_usage.py:1]: remove space after shebang

Comment (Avgor46) [reply update_usage.py]: done

File: update_usage.py (hunk 2 above)
Comment (Avgor46) [reply update_usage.py]: removed

**Review by hkctkuy — APPROVED** [review]

**Review by hkctkuy — CHANGES_REQUESTED** [review]
Add script usage to contributing pipeline

**Comment by anfedotoff** [comment]
Check that script is not included in release package on crates.io

File: update_usage.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        number_of_sections = len(splitted)
+        if number_of_sections != 4 and number_of_sections != 3:
+            print('Bad format in help message: ' + p)
+            continue
+        for i in range(1, number_of_sections):
+            splitted[i] = '\n'.join(['    ' + line for line in splitted[i].split('\n') if line])
+        new_message = '\n\n'.join(splitted) + '\n\n'
+        content = re.sub(f'## {p}\n\n' + '(.|\n)*?\n\n' * number_of_sections, \
+                         f'## {p}\n\n' + new_message, \
+                         content)
```
Comment (anfedotoff) [inline update_usage.py:35]: maybe add `if __name__ == "__main__"`?
Add describe some cases where we need manual updates of docs/usage.md

**Review by anfedotoff — CHANGES_REQUESTED** [review]

File: update_usage.py (hunk 4 above)
Comment (Avgor46) [reply update_usage.py]: done

I changed the regex so it should work in all cases.

**Review by hkctkuy — CHANGES_REQUESTED** [review]
Also update Contributing in README

**Review by hkctkuy — APPROVED** [review]

**Review by anfedotoff — APPROVED** [review]

**Pull request merged**